python-test-samples/apigw-lambda-dynamodb/tests/unit/mock_test.py

```diff
@@ -0,0 +1,135 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: MIT-0
+
+from os import environ
+import json
+from datetime import datetime
+from unittest import TestCase
+from typing import Any, Dict
+import yaml
+import boto3
+from boto3.dynamodb.conditions import Key
+import moto
+
+# Import the handler under test
+from src import app
+
+# Mock the DynamoDB Service during the test
+@moto.mock_dynamodb
+
+class TestSampleLambdaWithDynamoDB(TestCase):
+    """
+    Unit Test class for src/app.py
+    """
+    
+    def read_sam_template(self, sam_template_fn : str = "template.yaml" ) -> dict:
+        """
+        Utility Function to read the SAM template for the current project
+        """
+        with open(sam_template_fn, "r") as fp:
+            template =fp.read().replace("!","")   # Ignoring tags
+            try:
+                return yaml.safe_load(template)
+            except yaml.YAMLError as exc:
+                    print(exc)
+
+    def load_test_event(self, test_event_file_name: str) ->  Dict[str, Any]:
+        """
+        Load a sample event from a file
+        """
+        with open(f"tests/events/{test_event_file_name}.json") as f:
+            event = json.load(f)
+            # validate(event=event, schema=schemas.INPUT)
+            return event
+
+    def setUp(self) -> None:
+        """
+        Test Set up:
+           1. Create the lambda environment variale DYNAMODB_TABLE_NAME
+           2. Build a DynamoDB Table according to the SAM template
+           3. Populate DynamoDB Data into the Table for test
+        """
+
+        # Create a name for a test table, and set the environment
+        self.test_ddb_table_name = "unit_test_ddb_table_name"
+        environ["DYNAMODB_TABLE_NAME"] = self.test_ddb_table_name 
+
+        # Create a mock table using the definition from the SAM YAML template
+        sam_template_table_properties = self.read_sam_template()["Resources"]["DynamoDBTable"]["Properties"]
+        self.mock_dynamodb = boto3.resource("dynamodb", region_name="us-east-1")
+        self.mock_dynamodb_table = self.mock_dynamodb.create_table(
+                TableName = self.test_ddb_table_name,
+                KeySchema = sam_template_table_properties["KeySchema"],
+                AttributeDefinitions = sam_template_table_properties["AttributeDefinitions"],
+                BillingMode = sam_template_table_properties["BillingMode"]
```

Comment:
Just a heads up ... one limitation of this approach is that the use of intrinsics like `!If` or `!Ref` will break this.

Reply:
Adding a comment to call out the limitation of this simple approach in the next commit, and will add to the README as well.